File: mini_project/src/main/java/com/example/mini_project/entities/Job.java
package com.example.mini_project.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@NoArgsConstructor
@RequiredArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "services")
public class Job  implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name")
    @NonNull
    private String name;


    @Column(name = "description")
    @NonNull
    private String description;

    @Column(name = "avg_price")
    @NonNull
    private double price;

    @OneToOne
    @JoinColumn(name = "Image_id")
    private Image image;
    @OneToMany(mappedBy = "job")
    @JsonIgnore
    private List<Worker> workers;



}

File: mini_project_front/src/app/classes/job.ts
import { Image } from './image';
import { Worker } from './worker';

export class Job {
    id: number;
    name: string;
    description: string;
    price: number;
    workers!: Worker[];
    image!: Image;
    constructor(id: number, name: string, description: string, price: number, image: Image) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.image = image;
    }
}
